architecture/src/iosMain/kotlin/observable/Observable.kt

```diff
@@ -160,6 +161,28 @@ class FlowSubject<T>(private val flowable: HotFlowable<T>, private val coroutine
     }
 }
 
+/**
+ * [Subject] that synchronizes its value to a [MutableStateFlow]
+ * @param stateFlow The [MutableStateFlow] to synchronize to
+ * @param coroutineScope The [CoroutineScope] on which to observe changes to the [HotFlowable]
+ */
+class StateFlowSubject<T>(private val stateFlow: MutableStateFlow<T>, private val coroutineScope: CoroutineScope) : Subject<T>() {
+
+    init {
+        coroutineScope.launch(Dispatchers.Main) {
+            stateFlow.collect {
+                super.post(it)
+            }
+        }
+    }
+
+    override fun post(newValue: T) {
+        coroutineScope.launch(Dispatchers.Main) {
+            stateFlow.value = newValue
+        }
+    }
+}
```

Comment:
More of a comment, we are doing a lot of async, `launch`, `collect`,`post` 

We should probably at least try to use the `Dispatchers.Main.immidiate`. If you want I can create an issue to approach this more broadly however.

Reply:
Thats a good suggestion. That might actually fix the blinking issue @tstanik and @demarteau are experiencing in SplendoFit

Reply:
Apparently (at least on Android) that's due to Compose doing something similar (async) inside their own LiveData binding.